scripts/run_soupX.R

```diff
@@ -45,20 +46,134 @@ clusters_file <- opts$cluster
+# The default values place only a very weak constraint, as can be seen by setting a uniform prior.
+sc <- autoEstCont(sc) #priorRhoStdDev = 0.3
+print(sc)
+
+# Genes with highest expression in background. These are often enriched for ribosomal proteins.
+head(sc$metaData)
+head(sc$soupProfile[order(sc$soupProfile$est, decreasing = TRUE), ], n = 20)
+
+# To visualize the distribution of expression (relative to what would be expected were each cell pure background) across all cells in the data set. 
+# When no geneset is provided, the function will try and guess which genes might be useful.
+print(plotMarkerDistribution(sc))
+
+# Correcting expression profile
+# This will contain a corrected matrix to be used in place of the original table of counts in downstream analyses.
+# use roundToInt option to make sure we output integer matrix.
+out <- adjustCounts(sc, roundToInt = TRUE) 
+
+# Optional but we can keep original matrix
+# Attention as if we do, then for the next step we need to ensure we are using the correct assay, i.e., `RNA_SoupX`
+seurat_obj[["RNA_SoupX"]] <- CreateAssayObject(counts = seurat_obj@assays$RNA@counts)
+seurat_obj@assays$RNA@counts <- out
+
+# Setup assay for seurat object
+DefaultAssay(seurat_obj) <- "RNA_SoupX"
+seurat_obj
+
+
+#########################################################################################
+# Investigating changes in expression
+# Before proceeding let's have a look at what this has done. 
+# We can get a sense for what has been the most strongly decreased 
+# by looking at the fraction of cells that were non-zero now set to zero after correction.
+# Rename to toc or it gives NULL
+toc = filtered_matrix
+toc = sc$toc
+
+cntSoggy = rowSums(sc$toc > 0)
+cntStrained = rowSums(out > 0)
+mostZeroed = tail(sort((cntSoggy - cntStrained)/cntSoggy), n = 10)
+print(mostZeroed)
+
+
+# List of gene markers showing on this list are highly specific markers of one cell type or group of cells.
+# This is important to note as it may lead to erroneous inferences of potential cell specific genes.
+# For example, presence of mitochondrial genes MT-ND4, MT-ND4L or immune cells.
+
+# If on the other hand we focus on genes for which there is a quantitative difference,
+print(tail(sort(rowSums(sc$toc > out)/rowSums(sc$toc > 0)), n = 20))
+# Then we might notice different gene markers associated with other pathways and cell types.
+
+
+# Visualizing expression distribution
+# Way back at the start, we did a quick visualization to look at how the ratio of IGKC expression to pure soup was distributed. 
+# Now that we've corrected our data, we can see how that compares to our corrected data. 
+# The function plotChangeMap can help us with this. 
+# By default it plots the fraction of expression in each cell that has been deemed to be soup and removed.
+print(plotChangeMap(sc, out, "MT-ND4L")) #mtDNA
```

Comment:
Is this from a test dataset, or something universal to all single cell experiments?